tests/unit/s2n_self_talk_test.c

```diff
@@ -55,6 +55,8 @@ void mock_client(int writefd, int readfd)
 
     s2n_negotiate(conn, &blocked);
 
+    s2n_connection_free_handshake(conn);
```

Comment:
Do we need a test that we can wipe and reuse this connection for a new handshake after `s2n_connection_free_handshakes`?

Reply:
sounds good, will extend this test.